packages/node/src/core/evm/fulfillments/withdrawals.ts

```diff
@@ -0,0 +1,108 @@
+import { ethers } from 'ethers';
+import { go } from '../../utils/promise-utils';
+import * as logger from '../../logger';
+import * as wallet from '../wallet';
+import { ClientRequest, LogsErrorData, RequestStatus, TransactionOptions, Withdrawal } from '../../../types';
+
+type SubmitResponse = ethers.Transaction | null;
+
+export async function submitWithdrawal(
+  airnode: ethers.Contract,
+  request: ClientRequest<Withdrawal>,
+  options: TransactionOptions
+): Promise<LogsErrorData<SubmitResponse>> {
+  if (request.status !== RequestStatus.Pending) {
+    const logStatus = request.status === RequestStatus.Fulfilled ? 'DEBUG' : 'INFO';
+    const log = logger.pend(
+      logStatus,
+      `Withdrawal wallet index:${request.walletIndex} for Request:${request.id} not actioned as it has status:${request.status}`
+    );
+    return [[log], null, null];
+  }
+
+  const estimateTx = airnode.estimateGas.fulfillWithdrawal(
+    request.id,
+    request.providerId,
+    request.requesterIndex,
+    request.destinationAddress,
+    // We need to send some funds for the gas price calculation to be correct
```

Comment:
The receiving address may be a `payable` contract that does one thing when it receives 1 Wei and  another thing when it receives 1 ETH so this estimation is not guaranteed to be correct. I'm being pedantic though, this is a silly thing to be concerned about.
I guess the guaranteed way of dealing with is having the user provider the gas limit and gas price, but that's terrible UX.

Reply:
That's not a problem at all. I can just shuffle the code around so that the balance is fetched first then that is used as the value (before subtracting the transaction cost). I guess that would be more accurate than 1 Wei.

Good catch on the `walletIndex` references.

Reply:
If you don't subtract it it would probably revert. I think leaving it as is will be cleaner.